.github/workflows/build-dumb-pypi.yml

```diff
@@ -0,0 +1,177 @@
+      ansible-base-ref:
+        default: devel
+        description: The GitHub repo ref for pulling ansible-base from
+        required: true
+      ansible-build-data-slug:
+        default: ansible-community/ansible-build-data
+        description: The GitHub repo slug for pulling ansible build data from
+        required: true
+      ansible-build-data-ref:
+        default: main
+        description: The GitHub repo ref for pulling ansible build data from
+        required: true
+  schedule:
+  # https://crontab.guru/#23_*/4_*_*_*
+  - cron: 23 */4 * * *  # At minute 23 past every 4th hour.
+
+jobs:
+  build-dumb-pypi:
+    name: Build ACD and ansible-base nightlies
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Switch to Python 3.8
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Figure out the source repos
+      id: build-settings
+      run: |
+        print(
+            '::set-output name=ansible-base-slug::${{
+                github.event_name == 'workflow_dispatch' &&
+                github.event.inputs.ansible-base-slug ||
+                'ansible/ansible'
+            }}'
+        )
+        print(
+            '::set-output name=ansible-base-ref::${{
+                github.event_name == 'workflow_dispatch' &&
+                github.event.inputs.ansible-base-ref ||
+                'devel'
+            }}'
+        )
+        print(
+            '::set-output name=ansible-build-data-slug::${{
+                github.event_name == 'workflow_dispatch' &&
+                github.event.inputs.ansible-build-data-slug ||
+                'ansible-community/ansible-build-data'
+            }}'
+        )
+        print(
+            '::set-output name=ansible-build-data-ref::${{
+                github.event_name == 'workflow_dispatch' &&
+                github.event.inputs.ansible-build-data-ref ||
+                'main'
+            }}'
+        )
+      shell: python
+    - name: Check out antsibull src from Git
+      uses: actions/checkout@v2
+    - name: >-
+        Check out ${{
+          steps.build-settings.outputs.ansible-build-data-slug
+        }}@${{
+          steps.build-settings.outputs.ansible-build-data-ref
+        }} src from Git
+      uses: actions/checkout@v2
+      with:
+        path: ansible-build-data
+        ref: ${{ steps.build-settings.outputs.ansible-build-data-ref }}
+        repository: ${{ steps.build-settings.outputs.ansible-build-data-slug }}
+    - name: Make 'dist/' dir
+      run: |
+        from pathlib import Path
+        Path('dist').mkdir()
+      shell: python
+    - name: Install packaging-related dists from PyPI
+      run: >-
+        python -m
+        pip install
+        -U --user
+        pip setuptools wheel
+    - name: Install antsibull dist from PyPI
+      run: python -m pip install -U --user antsibull
+    - name: Install antsibull dist from GitHub
+      run: >-
+        python -m
+        pip install
+        -U --user
```

Comment:
o/
I'm still learning about the ansible releasing process and github actions but had a question: I count three different pip install tasks -- could there be a single one and could it pull from a requirements.txt file instead ?

Reply:
Usually, my preference is to have explicit tasks in CI reported distinctly. Just like tasks in Ansible: with their own names. This way, it's easier to identify what exactly goes wrong vs having to skim through many lines of logs if something explodes in the middle. It's like having a more precise context.

Also, since it's something that runs only in GHA, I chose to encapsulate everything in one file so that it's easy to see things together when reading it. If it was something runnable locally too or in different envs, I'd go for lockfiles (probably managed by pip-tools + dependabot).